### DISCUSSION TITLE
Repository metadata official definition and/or documentation

### DISCUSSION THREAD

**Discussion by carlosxmoreno** [community/community#122120] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hi all,
I am working on a research project related to project effort estimation, using github open repository metadata for the analysis phase.
I am looking for any official source of definition for github repository metadata, in order to work with a precise meaning for all attributes in the metadata.
Could anyone please point me in the right direction? I’ve tried already Github docs.

Related to this, my second question is if there is a better way of accessing github repository metadata without the need of downloading everything via API.

Thanks in advance, and best regards

**Comment by evelynforkhands** [comment, marked as answer]
Not sure about official definitions, but you could look at projects that specifically deal with repo metadata. For example, [this Github Action](https://github.com/marketplace/actions/github-repository-metadata) that exposes a repo's metadata as env variables has quite a few definitions.

**Comment by tstran155** [comment]
Hi carlosxmoreno,

For the GitHub repository metadata, below are a few more resources you can check out:

- GitHub API Documentation like GitHub API v3 (https://developer.github.com/v3/) and GitHub API v4 (GraphQL)(https://developer.github.com/v4/)
- GitHub Glossary: https://docs.github.com/en/github/getting-started-with-github/github-glossary

For the API access to GitHub repository metadata, here are a few things you can try:

- Instead of downloading all metadata at once, you can make specific API calls to fetch only the pieces of data (number of commits, contributors, or issues) you need
- GitHub’s GraphQL API allows you to tailor requests to retrieve exactly what you need. This can be simpler compared to the REST API where you might receive more data than necessary.
- Web scraping is also a viable option. 
- Other tools and services that aggregate GitHub data and might provide more efficient access to metadata. You might want to try GH Archive (https://www.gharchive.org/) or Google BigQuery’s GitHub dataset (https://cloud.google.com/blog/topics/public-datasets/github-on-bigquery-analyze-all-the-open-source-code).

 Good luck!
